packages/cli-main/src/cli/services/upgrade.ts

```diff
@@ -0,0 +1,174 @@
+import {error, file, os, output, path, system} from '@shopify/cli-kit'
+import {
+  addNPMDependencies,
+  findUpAndReadPackageJson,
+  checkForNewVersion,
+  DependencyType,
+  getPackageManager,
+} from '@shopify/cli-kit/node/node-package-manager'
+
+// Canonical list of oclif plugins that should be installed globally
+const globalPlugins = ['@shopify/theme']
+
+interface PackageJsonContents {
+  name: string
+  dependencies?: {[name: string]: string}
+  devDependencies?: {[name: string]: string}
+  oclif?: {
+    plugins?: string[]
+  }
+}
+
+export async function upgrade(directory: string, currentVersion: string): Promise<void> {
+  let newestVersion: string | void
+
+  const projectDir = await getProjectDir(directory)
+  if (projectDir) {
+    newestVersion = await upgradeLocalShopify(projectDir, currentVersion)
+  } else if (usingPackageManager()) {
```

Comment:
I find this clause a bit confusing. What does it mean that the `npm_config_user_agent` environment variable is not present? Who is setting that variable and when?

Reply:
It's a [standard variable](https://docs.npmjs.com/cli/v8/using-npm/config#user-agent) set by the package managers. If it's there, we assume this is in a project and therefore the upgrade is local. Otherwise we're assuming a global upgrade.